generator/tree/traversal.py

```diff
@@ -60,7 +60,7 @@ def _iterate(self):
                 discovered.add(node)
                 stack.append(State(node=node, processed=True))
 
-                for child in _Traversal.children(node):
+                for child in reversed(_Traversal.children(node)):
```

Comment:
Could you please add a small comment why this is needed, since it not entirely obvious. Thanks!